src/filesystem.rs

```diff
@@ -100,27 +93,65 @@ impl<R: SquashFsReader> FilesystemReader<R> {
     }
 }
 
-struct FilesystemFileReader<'a, R: SquashFsReader> {
-    filesystem: &'a FilesystemReader<R>,
-    file: &'a BasicFile,
-    last_read: Vec<u8>,
-    //current block, after all blocks maybe there is a fragment, None is finished
-    current_block: Option<usize>,
-    bytes_available: usize,
-    pos: u64,
-}
+struct FilesystemFileReader<'a, R: SquashFsReader>(Option<InnerFilesystemFileReader<'a, R>>);
 impl<'a, R: SquashFsReader> FilesystemFileReader<'a, R> {
     pub fn new(filesystem: &'a FilesystemReader<R>, file: &'a BasicFile) -> Self {
-        Self {
+        Self(Some(InnerFilesystemFileReader {
             filesystem,
             file,
             last_read: vec![],
             current_block: Some(0),
             bytes_available: file.file_size as usize,
             pos: file.blocks_start.into(),
+        }))
+    }
+}
+impl<'a, R: SquashFsReader> Read for FilesystemFileReader<'a, R> {
+    fn read(&mut self, buf: &mut [u8]) -> std::io::Result<usize> {
+        let inner = if let Some(inner) = &mut self.0 {
```

Comment:
```suggestion
        let Some(inner) = &mut self.0 else {
```

Reply:
This is unstable in rust 1.64

Reply:
If you plan to change the rust version, maybe this should be pushed into other PR and include the change to [Cargo](https://github.com/wcampbell0x2a/backhand/blob/master/Cargo.toml)

Reply:
Oh yep, completely forgot the msrv.